src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -34,21 +31,18 @@
 class ItemDataProvider implements ItemDataProviderInterface
 {
     private $managerRegistry;
-    private $propertyNameCollectionFactory;
-    private $propertyMetadataFactory;
+    private $identifierManager;
     private $itemExtensions;
 
     /**
-     * @param ManagerRegistry                        $managerRegistry
-     * @param PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory
-     * @param PropertyMetadataFactoryInterface       $propertyMetadataFactory
-     * @param QueryItemExtensionInterface[]          $itemExtensions
+     * @param ManagerRegistry               $managerRegistry
+     * @param IdentifierManagerInterface    $identifierManager
+     * @param QueryItemExtensionInterface[] $itemExtensions
      */
-    public function __construct(ManagerRegistry $managerRegistry, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, array $itemExtensions = [])
```

Comment:
Nice link! To bad though, this is a nice improvement IMO.

I can make this work without BC by still injecting both metadata factories and calling `new IdentifierManager` manually in the constructor, not sure that it'd be accepted though.